components/proxy/src/main/kotlin/com/hotels/styx/ObjectTags.kt

```diff
@@ -15,11 +15,42 @@
  */
 package com.hotels.styx
 
+private const val LBGROUP = "lbGroup"
+private val LBGROUP_REGEX = "$LBGROUP=(.+)".toRegex()
 fun lbGroupTag(name: String) = "lbGroup=$name"
-
-fun lbGroupTagValue(tag: String): String? = "lbGroup=(.+)".toRegex()
-        .matchEntire(tag)
+fun lbGroupTag(tags: Set<String>) = tags.firstOrNull(::isLbGroupTag)
+fun isLbGroupTag(tag: String) = LBGROUP_REGEX.matches(tag)
+fun lbGroupTagValue(tags: Set<String>) = lbGroupTagValue(lbGroupTag(tags)?:"")
+fun lbGroupTagValue(tag: String): String? = LBGROUP_REGEX.matchEntire(tag)
         ?.groupValues
         ?.get(1)
 
 fun sourceTag(creator: String) = "source=$creator"
+
+private const val STATE = "state"
+const val STATE_ACTIVE = "active"
+const val STATE_UNREACHABLE = "unreachable"
+const val STATE_CLOSED = "closed"
+private val STATE_REGEX = "$STATE=(.+)".toRegex()
+fun stateTag(value: String) = "$STATE=$value"
+fun stateTag(tags: Set<String>) = tags.firstOrNull(::isStateTag)
+fun isStateTag(tag: String) = STATE_REGEX.matches(tag)
+fun stateTagValue(tags: Set<String>) = stateTagValue(stateTag(tags)?:"")
+fun stateTagValue(tag: String) = STATE_REGEX.matchEntire(tag)
+        ?.groupValues
+        ?.get(1)
+
+private const val HEALTH = "health"
+const val HEALTH_SUCCESS = "success"
+const val HEALTH_FAIL = "fail"
```

Comment:
"Object health" is an ongoing state. But `success`/`fail` imply an outcome of some action. Therefore consider renaming tag values to `healthy`/`unhealthy`, or `reachable`/`unreachable`. Etc.

Reply:
These are the outcomes of actions, really, I suppose. Perhaps the "health" name is wrong. How about `healthcheck=[passing|failing]:N` ?